### DISCUSSION TITLE
Beginner problem

### DISCUSSION THREAD

**Discussion by Rahul251952** [Cyfrin/foundry-full-course-cu#2965] [General]
I am learning solidity since 2 months and i think i don't learn anything. please someone suggest me what i should do also i am learning foundry but can't get anything. please help.

**Comment by EngrPips** [comment]
Hello, You should make sure you understand every lesson before you move to the next one, and after you are confident you understand everything taught in a particular lesson, try to redo everything taught in that lesson all by yourself without checking the course resources and see how well you will do

**Reply by Rahul251952** [reply]
Thanks, Also tell me both courses of foundry is compulsory.

**Reply by EngrPips** [reply]
I will say every Foundry content is compulsory if you choose to learn Foundry.
